guides/hosting/configurations/shopware/staging.md

```diff
@@ -0,0 +1,213 @@
+        mailing:
+            # Disables the sending of mails (default: true)
+            disable_delivery: true
+        storefront:
+            # Shows a banner in the storefront when staging mode is active (default: true)
+            show_banner: true
+        administration:
+            # Shows a banner in the administration when staging mode is active (default: true)
+            show_banner: true
+        sales_channel:
+            domain_rewrite:
+                # See below for more information
+        elasticsearch:
+            # Checks that no indices are existing yet (default: true)
+            check_for_existence: true
+```
+
+One of the most important options is the `domain_rewrite`. This option allows you to rewrite the URLs to the staging domain. This allows multiple ways to rewrite the URLs:
+
+Using direct match (`equal`)
+
+
+```yaml
+# <shopware-root>/config/packages/staging.yaml
+shopware:
+    staging:
+        sales_channel:
+            domain_rewrite:
+                - type: equal
+                  match: https://my-live-store.com
+                  replace: https://my-staging-store.com
+                - # ... second rule
+```
+
+This compares the Sales Channel URL, and when it's equal to `https://my-live-store.com`, it will be replaced with `https://my-staging-store.com`.
+
+Replace using prefix (`prefix`)
+
+```yaml
+# <shopware-root>/config/packages/staging.yaml
+shopware:
+    staging:
+        sales_channel:
+            domain_rewrite:
+                - type: prefix
+                  match: https://my-live-store.com
+                  replace: https://my-staging-store.com
+                - # ... second rule
+```
+
+The difference here to the `equal` type is that it will only replace the URL when it starts with `https://my-live-store.com`, so all paths to that beginning will be replaced.
+
+`https://my-live-store.com/en` will be replaced with `https://my-staging-store.com/en`
+
+Replace using regex (`regex`)
+
+```yaml
+# <shopware-root>/config/packages/staging.yaml
+shopware:
+    staging:
+        sales_channel:
+            domain_rewrite:
+                - type: regex
+                  match: '/https?:\/\/(\w+)\.(\w+)$/m'
+                  replace: 'http://$1-$2.local'
+                - # ... second rule
+```
+
+This will use the regex to replace the URL. The match and replace are regular expressions. In this example a `https://my-live-store.com` will be replaced with `http://my-live-store.local`.
+
+### Usage of Apps
+
+The staging command will delete all apps with an active connection to an external service. This will be done to avoid data corruption or leaks in the live environment, as the staging environment is a copy of the live environment, so they keep a connection. After executing the command, you can install the App again, creating a new instance ID so the app will think it's an entirely different shop. In that way, the app installation is completely isolated from the live environment.
```

Comment:
```suggestion
The staging command will delete all apps with an active connection to an external service. This will be done to avoid data corruption or leaks in the live environment, as the staging environment is a copy of the live environment, so they keep a connection. After executing the command, you can install the App again, creating a new instance ID, so the app will think it's an entirely different shop. In that way, the app installation is completely isolated from the live environment.
```